### PR TITLE
[new release] albatross (1.0.0)

### PR DESCRIPTION
**Pull request by hannesm**
Albatross - orchestrate and manage MirageOS unikernels with Solo5

- Project page: <a href="https://github.com/roburio/albatross">https://github.com/roburio/albatross</a>

##### CHANGES:

Initial release

### CODE REVIEW

File: packages/albatross/albatross.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  "metrics-influx" {>= "0.2.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depexts: [
+  ["libnl-3-dev" "libnl-route-3-dev"] {os-family = "debian"}
+  ["libnl3" "libnl3-devel"] {os-family = "centos"}
+]
```
Comment (kit-ty-kate) [inline packages/albatross/albatross.1.0.0/opam:44]: Would it be possible to move this into a separate `conf-libnl` package?

Comment (reynir) [reply packages/albatross/albatross.1.0.0/opam:44]: I will do that.

libnl3 is divided into a handful of libraries. On debian they are packaged separately while on centos they seem to all live in libnl3-devel. Would you advice creating conf-libnl3, conf-libnl-route3, etc. or just one conf-libnl3 with all of them?

Comment (hannesm) [reply packages/albatross/albatross.1.0.0/opam:44]: thanks @reynir -- I suggest a single `conf-libnl3` which includes the needed packages. when some other package needs a subset, we can always split conf-libnl3 into multiple packages.

**Comment by hannesm** [comment]
I rebased and now use the conf-libnl3 package (#17745 by @reynir).

**Comment by kit-ty-kate** [comment]
Is it expected to fail on ppc64?
```
#=== ERROR while compiling albatross.1.0.0 ====================================#
# context              2.0.7 | linux/ppc64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/roburio/albatross/releases/download/v1.0.0/albatross-v1.0.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/albatross.1.0.0
# command              ~/.opam/4.11/bin/dune build -p albatross -j 79
# exit-code            1
# env-file             ~/.opam/log/albatross-1-6a4d96.env
# output-file          ~/.opam/log/albatross-1-6a4d96.out
### output ###
#     discover stats/c_flags.sexp,stats/c_library_flags.sexp (exit 2)
# (cd _build/default/stats && config/discover.exe)
# Fatal error: exception Failure("Unsupported platform: elf")
```
If yes, could you mark them as unavailable the same way it is done for solo5?

**Comment by kit-ty-kate** [comment]
There are also similar failures on x86_32 and arm32 (arm64 is fine)

Otherwise there is another failure on Ubuntu 20.10, Alpine and Archlinux:
```
#=== ERROR while compiling albatross.1.0.0 ====================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/roburio/albatross/releases/download/v1.0.0/albatross-v1.0.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/albatross.1.0.0
# command              ~/.opam/4.11/bin/dune build -p albatross -j 127
# exit-code            1
# env-file             ~/.opam/log/albatross-1-601178.env
# output-file          ~/.opam/log/albatross-1-601178.out
### output ###
#          gcc stats/albatross_stats_stubs.o (exit 1)
# (cd _build/default/stats && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -I/usr/include/libnl3 -I/usr/include/libnl3 -g -I /home/opam/.opam/4.11/lib/ocaml -I /home/opam/.opam/4.11/lib/asn1-combinators -I /home/opam/.opam/4.11/lib/astring -I /home/opam/.opam/4.11/lib/bigarray-compat -I /home/opam/.opam/4.11/lib/bos -I /home/opam/.opam/4.11/lib/bytes -I /home/opam/.opam/4.11/lib/checkseum -I /home/opam/.opam/4.11/lib/cstruct -I /home/opam/.opam/4.11/lib/decompress -I /home/opam/.opam/4.11/lib/decompress/impl -I /home/opam/.opam/4.11/lib/domain-name -I /home/opam/.opam/4.11/lib/duration -I /home/opam/.opam/4.11/lib/fmt -I /home/opam/.opam/4.11/lib/fpath -I /home/opam/.opam/4.11/lib/ipaddr -I /home/opam/.opam/4.11/lib/jsonm -I /home/opam/.opam/4.11/lib/logs -I /home/opam/.opam/4.11/lib/lwt -I /home/opam/.opam/4.11/lib/lwt/unix -I /home/opam/.opam/4.11/lib/macaddr -I /home/opam/.opam/4.11/lib/metrics -I /home/opam/.opam/4.11/lib/mmap -I /home/opam/.opam/4.11/lib/ocaml/threads -I /home/opam/.opam/4.11/lib/ocplib-endian -I /home/opam/.opam/4.11/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.11/lib/optint -I /home/opam/.opam/4.11/lib/ptime -I /home/opam/.opam/4.11/lib/ptime/os -I /home/opam/.opam/4.11/lib/result -I /home/opam/.opam/4.11/lib/rresult -I /home/opam/.opam/4.11/lib/seq -I /home/opam/.opam/4.11/lib/stdlib-shims -I /home/opam/.opam/4.11/lib/uchar -I /home/opam/.opam/4.11/lib/uutf -I /home/opam/.opam/4.11/lib/zarith -I ../src -o albatross_stats_stubs.o -c albatross_stats_stubs.c)
# albatross_stats_stubs.c:14:10: fatal error: sys/sysctl.h: No such file or directory
#    14 | #include <sys/sysctl.h>
#       |          ^~~~~~~~~~~~~~
# compilation terminated.
```

**Comment by hannesm** [comment]
thanks @kit-ty-kate -- I pushed another commit to hopefully fix these errors

**Comment by hannesm** [comment]
the remaining failure on alpine is `sys/sysctl.h` does not exist -- according to https://github.com/supertuxkart/stk-code/issues/4172 this is `linux/sysctl.h` on alpine. I'll fix this eventually upstream, should I put an availability constraint (`os-distribution != "alpine"`)?

**Comment by kit-ty-kate** [comment]
Apparently `sys/sysctl.h` is deprecated in glibc too. I fixed it in https://github.com/roburio/albatross/pull/58, this should fix everything.

Adding `os-distribution != "alpine"` wouldn't be enough since ubuntu 20.10 and (maybe) archlinux are also affected.

**Comment by hannesm** [comment]
sounds good, I'll close this PR and cut a new release with the changes merged

**Pull request closed without merging**